### ISSUE TITLE
[Accessibility]: unable to focus on theme Select in user dropdown menu

### ISSUE THREAD

**Issue by oleggromov** [sourcegraph/sourcegraph#33262]
### Problem description

In the user dropdown menu, theme selector never gets selected when navigating using keyboard.

https://user-images.githubusercontent.com/2196347/161015125-888e0b75-db10-4170-be62-bb3297a6188d.mov

### Expected behavior

When using keyboard arrow keys to navigate between menu items, the theme selector should be focused on and allowed to use keyboard to choose an option from the dropdown.

It is unclear, however, how to get into the "focusing context" of the select box and back to menu's one.

### Additional details

This had been briefly discussed in another PR: https://github.com/sourcegraph/sourcegraph/pull/33134#issuecomment-1083313789

There's a few issues preventing us from solving it in a straightforward manner:

1. If we wrap the select box in a `MenuItem`, it would not pass clicks (causing its `onSelect` to fire) to the select.
2. `ReachMenuItem` doesn't provide a way to prevent this default behavior as `onSelect` callback isn't passed an event object or anything like that.

We will likely need to change the `MenuItem` implementation to circumvent the issue.

**Comment by umpox** [comment]
Related audit: https://github.com/sourcegraph/sourcegraph/issues/34415
